openjdk/sun/nio/ch/FileChannelImpl.java

```diff
@@ -1068,7 +1068,7 @@ private static long mapViewOfFileWin32(FileStream fs, int prot, long position, l
             IntPtr p = MapViewOfFile(hFileMapping, mapAccess, (int)(position >> 32), (int)position, IntPtr.op_Explicit(length));
             err = cli.System.Runtime.InteropServices.Marshal.GetLastWin32Error();
             hFileMapping.Close();
-            if (p.Equals(IntPtr.Zero))
+            if (p.Equals((cli.System.Object)IntPtr.Zero))
```

Comment:
Just curiosity - why does IntPtr require the cast?

Reply:
Well, this is a workaround to the bad IL problem.

In the framework, IntPtr extends Object (eventually).  In the core it also implements IEquatable.  Because of this IKVM compiler generates a bad code calling IEquatable.Equals.  C# compiler (compiling for the core) produces the Object.Equals call.  The simple cast does a good job for the IKVM compiler.

I'm not confident enough (at least yet) to go into the IKVM code generation and make changes there.  So, I chose to apply this simpler workaround.

Reply:
I suggest putting that explanation in as a comment

Reply:
I made a comment in the code.